autoextract_poet/providers.py

```diff
@@ -0,0 +1,49 @@
+from typing import Type
+
+from scrapy.http import Request
+from scrapy.statscollectors import StatsCollector
+from scrapy_poet.page_input_providers import (
+    PageObjectInputProvider,
+    provides,
+)
+
+from autoextract_poet.inputs import (
+    AutoExtractProductResponseData,
+    AutoExtractProductListResponseData,
+)
+from autoextract_poet.utils import request
+
+
+class AutoExtractProvider(PageObjectInputProvider):
+
+    page_type: str
+    page_input_class: Type
+
+    def __init__(self, request: Request, stats: StatsCollector):
+        self.request = request
+        self.stats = stats
+
+    async def __call__(self):
+        try:
+            # FIXME: how to configure if you want full HTML or not?
+            data = await request(self.request.url, self.page_type)
+        except Exception:
+            self.stats.inc_value(f"autoextract/{self.page_type}/error")
+            raise
+
+        self.stats.inc_value(f"autoextract/{self.page_type}/success")
+        return self.page_input_class(data=data)
+
+
+@provides(AutoExtractProductResponseData)
+class AutoExtractProductResponseDataProvider(AutoExtractProvider):
+
```

Comment:
Maybe rename to `autoextract_page_type`.

Reply:
I guess the context gives the idea of autoextract

Reply:
I've renamed class names.